### PR TITLE
feat(modal-checkout): update modal checkout checkout form with newspack ui

### PR DESCRIPTION
**Pull request by chickenn00dle**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of `1206197591054328/1206405274735557/f`

This PR updates the remainder of the checkout form with newspack ui including:

The billing info form:
Screenshot 2024-01-30 at 17 15 39

The checkout form:
Screenshot 2024-01-30 at 17 17 29

The variation selection form:
Screenshot 2024-01-30 at 17 19 07

### How to test the changes in this Pull Request:

Important! Before testing, make sure you have https://github.com/Automattic/newspack-plugin/pull/2878 checked out as many of the woocommerce form styles have been overridden there.

1. On your test site, add a donation block, a checkout button block for variation product, and another checkout button block for a subscription product to any page.
2. Also ensure you've enabled coupons in WooCommerce settings AND have installed [WooCommerce Subscriptions Gifting](https://github.com/woocommerce/woocommerce-subscriptions-gifting)
3. On the front end, visit this page and select one of the checkout buttons of the donation block.
4. Confirm you see the updated ui in the billing and checkout portions of the checkout flow (pictured above)
5. Repeat the above steps for the checkout block for the variable product.
6. Confirm you see the updated ui in the variation, billing, and checkout portions of the checkout flow (pictured above). Lookout for the coupon field in the checkout portion of the flow:
Screenshot 2024-01-30 at 17 28 47
8. Repeat again for the checkout block for the subscription product.
9. Confirm you see the updated ui in the billing and checkout portions of the checkout flow (pictured above). Look out for the subscription gift toggle in the billing info section of the flow:
Screenshot 2024-01-30 at 17 28 37
10. Make sure to smoke test the form, submitting with missing or invalid inputs, etc.

**Testing the standalone modal checkout**

We also want to make sure the checkout block doesn't appear too broken when newspack ui is not present. We don't need it to match, but the checkout flow should use a combination of default WC and a bit of custom newspack block styles.

1. Check out `master` over in the `newspack-plugin` repo so newspack ui is no longer present.
2. Go through the testing instructions above and make sure nothing looks janky 😰 

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206405274735557

### CODE REVIEW

**Comment by chickenn00dle** [comment]
Hey @laurelfulford, I hope it's okay if I ping you for this one as well since you are the most familiar with how the newspack ui styles should look and are already looking at https://github.com/Automattic/newspack-blocks/pull/1649. 

Let me know if you've got too much on your plate though!

**Review by dkoo — CHANGES_REQUESTED** [review]
@chickenn00dle looking mostly good, as far as I can tell! Noting the broken "Sign In" buttons, but I see your `TODO` comment about this. Other than that, I saw some minor styling issues mainly in the payment gateway section.

Some of the font sizing seems off. In this screenshot, the text that starts "Pay with your credit card via Stripe" and the privacy notice below ("Your personal data will be used to process your order...") look pretty minuscule. Also, the checkbox labels look unnaturally large.

Screenshot 2024-01-31 at 2 21 16 PM


Refer to this screenshot from the design mocks for relative type scale. Also note that the design mock includes both a bold and non-bold element for the "cover fees" checkbox. Currently our settings only allow for a single text string, so it might be hard to emulate this right now, but let's make sure the styles support the separate lines.

Screenshot 2024-01-31 at 11 52 53 AM

File: includes/class-modal-checkout.php [hunk 1]
```diff
@@ -1238,5 +1244,22 @@ public static function jetpack_active_modules( $modules ) {
...
+	 * @param string $html The button html.
+	 */
+	public static function order_button_html( $html ) {
+		if ( ! self::is_modal_checkout() ) {
+			return $html;
+		}
+
+		$classname = self::get_class_prefix();
+
+		$newspack_ui_html = preg_replace( '/class=".*?"/', "class='{$classname}__button {$classname}__button--primary {$classname}__button--wide'", $html );
```
[the remark this answers is not in the document]
Comment (chickenn00dle) [reply includes/class-modal-checkout.php]: Good call! I've updated this in https://github.com/Automattic/newspack-blocks/pull/1653/commits/9f7ad1c898d747e7def0b390b738d4510b9793e3

**Comment by chickenn00dle** [comment]
Thanks for the review @dkoo!

> Some of the font sizing seems off. In this screenshot, the text that starts "Pay with your credit card via Stripe" and the privacy notice below ("Your personal data will be used to process your order...") look pretty minuscule. Also, the checkbox labels look unnaturally large.

Good catch. I've updated the text sizes in the newspack ui overrides over in https://github.com/Automattic/newspack-plugin/pull/2878 so they should be closer to the figma:

Screenshot 2024-02-01 at 13 51 54

> Also note that the design mock includes both a bold and non-bold element for the "cover fees" checkbox. Currently our settings only allow for a single text string, so it might be hard to emulate this right now, but let's make sure the styles support the separate lines.

Yeah. I noticed that too. I'm a bit hesitant to add styling for this until we adjust our settings to allow for multiple strings of text here. Ultimately, I think it will be pretty simple once we do though. Happy to go ahead and add the styles now though!

**Review by dkoo — APPROVED** [review]
This is looking better to me, now! Thanks for the changes.
Screenshot 2024-02-02 at 6 34 43 PM

**Pull request merged**